File: app/src/main/java/com/orangepenguin/boilerplate/util/AndroidSharedPreferenceUtil.java
package com.orangepenguin.boilerplate.util;

import android.content.Context;

public class AndroidSharedPreferenceUtil implements SharedPreferencesUtil {

    private static final String FILE_NAME = "APPLICATION_PREFERENCES";
    private final Context context;

    public AndroidSharedPreferenceUtil(Context context) {
        this.context = context;
    }

    @Override
    public String getPreference(String key, String defaultValue) {
        return context.getSharedPreferences(FILE_NAME, Context.MODE_PRIVATE)
                .getString(key, defaultValue);
    }

    @Override
    public void savePreference(String key, String value) {
        context.getSharedPreferences(FILE_NAME, Context.MODE_PRIVATE).edit().putString(key, value).apply();
    }

    @Override
    public void clearPreference(String key) {
        context.getSharedPreferences(FILE_NAME, Context.MODE_PRIVATE).edit().remove(key).apply();
    }
}

File: app/src/main/java/com/orangepenguin/boilerplate/util/AndroidSharedPreferenceUtil.kt
package com.orangepenguin.boilerplate.util

import android.content.Context

class AndroidSharedPreferenceUtil(private val context: Context) : SharedPreferencesUtil {

    override fun getPreference(key: String, defaultValue: String?): String? =
            context.getSharedPreferences(FILE_NAME, Context.MODE_PRIVATE).getString(key, defaultValue)

    override fun savePreference(key: String, value: String) =
            context.getSharedPreferences(FILE_NAME, Context.MODE_PRIVATE).edit().putString(key, value).apply()

    override fun clearPreference(key: String) =
            context.getSharedPreferences(FILE_NAME, Context.MODE_PRIVATE).edit().remove(key).apply()

    companion object {
        private val FILE_NAME = "APPLICATION_PREFERENCES"
    }
}
